projects/storefrontlib/src/events/product/product-page-event.builder.ts

```diff
@@ -101,20 +95,24 @@ export class ProductPageEventBuilder {
   protected buildSearchPageResultsEvent(): Observable<SearchPageResultsEvent> {
     const searchResults$ = this.productSearchService.getResults().pipe(
       // skipping the initial value, and preventing emission of the previous search state
-      skip(1),
-      map((searchResults) => ({
-        searchTerm: searchResults.freeTextSearch,
-        numberOfResults: searchResults.pagination.totalResults,
-      })),
-      map((searchPage) => createFrom(SearchPageResultsEvent, searchPage))
+      skip(1)
     );
 
-    const searchPageEvent$ = this.eventService
-      .get(PageEvent)
-      .pipe(map((pageEvent) => pageEvent.semanticRoute === 'search'));
+    return this.eventService.get(PageEvent).pipe(
+      switchMap((pageEvent) => {
+        if (pageEvent.semanticRoute !== 'search') {
+          return EMPTY;
+        }
 
-    return searchPageEvent$.pipe(
-      switchMap((isSearchPage) => (isSearchPage ? searchResults$ : EMPTY))
+        return searchResults$.pipe(
+          map((searchResults) => ({
+            ...pageEvent,
+            searchTerm: searchResults.freeTextSearch,
+            numberOfResults: searchResults.pagination.totalResults,
```

Comment:
I'd like to see the `numberOfResults` in the category page results as well. Not sure if that would result in more inheritance of the model.

Reply:
Sure, done.